plugins/api/src/atomic_apply_metadata_operations.cpp

```diff
@@ -0,0 +1,559 @@
+        stmt.bind(1, _metadata.value.c_str());
+        stmt.bind(2, _metadata.units.c_str());
+
+        if (auto row = execute(stmt); row.next()) {
+            return row.get<int>(0);
+        }
+
+        return -1;
+    }
+
+    auto is_metadata_attached_to_object(nanodbc::connection& _db_conn, int _object_id, int _meta_id) -> bool
+    {
+        nanodbc::statement stmt{_db_conn};
+
+        prepare(stmt, "select count(*) from r_objt_metamap where object_id = ? and meta_id = ?");
+
+        stmt.bind(0, &_object_id);
+        stmt.bind(1, &_meta_id);
+
+        if (auto row = execute(stmt); row.next()) {
+            return row.get<int>(0) > 0;
+        }
+
+        return false;
+    }
+
+    auto insert_metadata(nanodbc::connection& _db_conn, const fs::metadata& _metadata) -> int
+    {
+        nanodbc::statement stmt{_db_conn};
+
+#ifdef ORA_ICAT
+        prepare(stmt, "insert into r_meta_main (meta_id, meta_attr_name, meta_attr_value, meta_attr_unit, create_ts, modify_ts) "
+                      "values (select r_objectid.nextval from dual, ?, ?, ?, ?, ?)");
+#else
+        prepare(stmt, "insert into r_meta_main (meta_id, meta_attr_name, meta_attr_value, meta_attr_unit, create_ts, modify_ts) "
+                      "values (nextval('r_objectid'), ?, ?, ?, ?, ?)");
+#endif // ORA_ICAT
+
+        const auto timestamp = std::to_string(std::chrono::system_clock::now().time_since_epoch().count());
+
+        stmt.bind(0, _metadata.attribute.c_str());
+        stmt.bind(1, _metadata.value.c_str());
+        stmt.bind(2, _metadata.units.c_str());
+        stmt.bind(3, timestamp.c_str());
+        stmt.bind(4, timestamp.c_str());
+
+        auto rows = execute(stmt);
+
+        //rodsLog(LOG_DEBUG, {{"function", __func__}, {"affected_rows", std::to_string(rows.affected_rows())}});
+
+        return get_meta_id(_db_conn, _metadata);
+    }
+
+    auto attach_metadata_to_object(nanodbc::connection& _db_conn, int _object_id, int _meta_id) -> void
+    {
+        nanodbc::statement stmt{_db_conn};
+
+        prepare(stmt, "insert into r_objt_metamap (object_id, meta_id, create_ts, modify_ts) "
+                      "values (?, ?, ?, ?)");
+
+        const auto timestamp = std::to_string(std::chrono::system_clock::now().time_since_epoch().count());
+
+        stmt.bind(0, &_object_id);
+        stmt.bind(1, &_meta_id);
+        stmt.bind(2, timestamp.c_str());
+        stmt.bind(3, timestamp.c_str());
+
+        auto rows = execute(stmt);
+
+        //rodsLog(LOG_DEBUG, "function", __func__}, {"affected_rows", std::to_string(rows.affected_rows())}});
+    }
+
+    auto detach_metadata_from_object(nanodbc::connection& _db_conn, int _object_id, int _meta_id) -> void
+    {
+        nanodbc::statement stmt{_db_conn};
+
+        prepare(stmt, "delete from r_objt_metamap where object_id = ? and meta_id = ?");
+
+        stmt.bind(0, &_object_id);
+        stmt.bind(1, &_meta_id);
+
+        auto rows = execute(stmt);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'rows' is assigned a value that is never used.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5179169)